src/traits.jl

```diff
@@ -53,13 +54,32 @@ function Size{SA <: StaticArray}(::Type{SA})
         """)
 end
 
-# Some @pure convenience functions.
 
-@pure get{S}(::Size{S}) = S
-@pure getindex{S}(::Size{S}, i::Int) = i <= length(S) ? S[i] : 1
+struct Length{L}
+    function Length{L}() where L
+        check_length(L)
```

Comment:
Oh, can you use that trick you were showing to avoid the need for `check_length`?